### PR TITLE
Add a few more `my` interfaces

### PR DESCRIPTION
**Pull request by madelinecameron**
Addresses https://github.com/karlicoss/HPI/issues/64

- Withings
- Level Health
- Eight Sleep
- ActivityWatch (local-only RescueTime, also write my own synchronization code so you can sync multiple computer's AW data)
- Typeform
- Todoist

Unfortunately the extractors are written in node then I wrote DALs for each in Python. Maybe I am being presumptive but I think exposing the idea that HPI doesn't have to be solely in python is fun. Love being able to leverage Python's data library ecosystem though, much better than node for this :joy:. 

 The `my` interfaces are pretty barebones and may not be up to standards, so happy to modify if there is a certain standard or direction you like better.

### CODE REVIEW

**Comment by seanbreckenridge** [comment]
Ah, I didn't realize you weren't watching [this repo](https://github.com/hpi/hpi), so you may have not seen the issue I raised [here](https://github.com/hpi/hpi/issues/1)

To expand on it a bit;

I used to maintain a direct fork/merge commits from this repo back into my own copy of HPI [here](https://github.com/seanbreckenridge/HPI), but that became more and more difficult to as time went on...

The solution for me was in the [SETUP.org](https://github.com/karlicoss/HPI/blob/master/doc/SETUP.org#addingmodifying-modules) file all along (described as the 'proper way')

I've since stopped merging commits that, and now install [both packages as editable](https://github.com/seanbreckenridge/HPI/blob/e618db9007a5bf6e2d0da2d459c956e04c1388ac/install#L42-L69), which means I keep up to date with this repository and can still add/overlay modules on top of that

i.e. you can have multiple `my/` directories installed at the same time, if they're installed as namespace packages (relevant lines in setup.py [here](https://github.com/karlicoss/HPI/blob/fb4924300508ce16accc3b79bd7253d127872712/setup.py#L17) and [here](https://github.com/seanbreckenridge/HPI/blob/master/setup.py#L4-L8)):

```
 $ python3
In [1]: from my.coding.commits import commits  # something from karlicoss/HPI

In [2]: from my.zsh import history  # something from my own repo
```

Another example of an overlay [here](https://github.com/karlicoss/hpi-personal-overlay)

Not saying these items can't be merged into here, thats up to karicoss, I've just found HPI to be more useful as a configuration/caching layer, giving me all the tools to load my data, but everyone uses different services/data sources, so I'd rather keep modules on my own branch. Otherwise, we'd eventually have hundreds of data sources in this one repository, and I don't know if anyone wants to maintain that :sweat_smile:

File: my/levelshealth.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+
+def inputs() -> Sequence[Path]:
+    return get_files(levelshealth.export_path)
+
+
+import levelshealth.dal as dal
+
+
+def glucoseScores():
```
Comment (karlicoss) [inline my/levelshealth.py:30]: if you annotate return types with `Iterator` (even `Iterator[Any]` should do), then `hpi doctor my.levelshealth` would discover them (so you won't need manual `stats` function)

File: my/levelshealth.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+'''
+Levels Health data
```
Comment (karlicoss) [inline my/levelshealth.py:2]: would be nice to include the link to the service in the docstring as well

File: my/levelshealth.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+REQUIRES = [
+    'git+https://github.com/hpi/levelshealth',
+]
+
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Sequence, Iterable
+
+from .core import Paths, get_files
```
Comment (karlicoss) [inline my/levelshealth.py:13]: we've recently been trying to start switching module code to fully qualified imports (`from my.core import...`) -- the reason is that eventually core will be split as a separate package, and then relative imports in modules won't make much sense
https://github.com/karlicoss/HPI/issues/102#issuecomment-796476003

File: my/eightsleep.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                }
+            except Exception as e:
+                # TODO use the trace? otherwise str() might be too short..
+                # r.g. for KeyError it only shows the missing key
+                # todo check for 'defensive'
+                d = {'error': f'{e} {s}'}
+            yield d
+    import pandas as pd # type: ignore
+    df = pd.DataFrame(it())
+    print(df)
```
Comment (karlicoss) [inline my/eightsleep.py:72]: best to avoid printing to stdout unless necessary

**Comment by karlicoss** [comment]
> Unfortunately the extractors are written in node then I wrote DALs for each in Python

No that's definitely cool, and I'm definitely pro using whatever tools people can use! 

> Not saying these items can't be merged into here, thats up to karlicoss, I've just found HPI to be more useful as a configuration/caching layer, giving me all the tools to load my data, but everyone uses different services/data sources, so I'd 
rather keep modules on my own branch. 

Yep, definitely suggest trying this approach out too!

> Otherwise, we'd eventually have hundreds of data sources in this one repository, and I don't know if anyone wants to maintain that 

Having too many modules is definitely an issue -- I dread the fate of the Spacemacs maintainer! I think for now I'm basically happy to merge things, stuff doesn't seem to break much; but need to start thinking of ways to making it more manageable in advance perhaps.

Although I would prefer if they had tests/ example data pointers somewhere, since otherwise I can't guarantee I won't break it :)

File: my/eightsleep.py (hunk 4 above)
Comment (madelinecameron) [reply my/eightsleep.py:72]: Ahhhh! Thanks! Didn't meant to leave this in.

**Pull request closed without merging**